### PR TITLE
[TensorIf] Allow to declare option without full dimension @open sesame 11/17 18:07

### PR DESCRIPTION
**Pull request by niley7464**
If  `NNS_TENSOR_RANK_LIMIT` is increased `~  ! tensor_if name=tif silent=false compared-value=A_VALUE compared-value-option=0:0:0:0,0 supplied-value=100 ! ~ `  can't be used.
This patch allows to declare compared-value-option in tensor_if without full dimension. 
Undeclared dimension will be filled with zero.

Signed-off-by: Yelin Jeong <yelini.jeong@samsung.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/nnstreamer/elements/gsttensor_if.c [hunk 1]
```diff
@@ -345,6 +345,49 @@ gst_tensor_if_set_property_glist (const GValue * value, GList ** prop_list,
...
+/**
+ * @brief Convert GValue to GList for cv option
+ */
+static void
+gst_tensor_if_set_property_cv_option (const GValue * value, GList ** prop_list)
+{
+  gint64 val;
+  gint length, i;
+  const gchar *param = g_value_get_string (value);
+  gchar **strv = g_strsplit_set (param, ",", -1);
```
Comment (songgot) [inline gst/nnstreamer/elements/gsttensor_if.c:357]: we need to use g_strfreev to free strv.
g_strfreev (strv);

File: gst/nnstreamer/elements/gsttensor_if.c [hunk 2]
```diff
@@ -345,6 +345,49 @@ gst_tensor_if_set_property_glist (const GValue * value, GList ** prop_list,
...
+    return;
+  }
+
+  g_value_init (&tmp, G_TYPE_STRING);
+  g_value_set_string (&tmp, strv[0]);
+
+  gst_tensor_if_set_property_glist (&tmp, prop_list, ":");
+
+  /* A_VALUE */
+  if (length == 2) {
```
Comment (songgot) [inline gst/nnstreamer/elements/gsttensor_if.c:374]: Are there any cases where the length is less than '2' ?
If we need to check if less than '2', we can fix it as below :
line 362 =>  if (length != 2)  
line 374 =>  remove if (length == 2) { }

Comment (niley7464) [reply gst/nnstreamer/elements/gsttensor_if.c:374]: If compared-value option is TENSOR_AVERAGE_VALUE, length will be 1.
So I just checked if length is 2 ( option A_VALUE ) to append zero values and n-th tensor info.

File: gst/nnstreamer/elements/gsttensor_if.c (hunk 1 above)
Comment (niley7464) [reply gst/nnstreamer/elements/gsttensor_if.c:357]: Added `g_strfreev (strv);`
Thank you :)

**Comment by anyj0527** [comment]
You may rebase this branch to the recent main branch to resolve 32bit gbs build failure

File: gst/nnstreamer/elements/gsttensor_if.c (hunk 2 above)
Comment (songgot) [reply gst/nnstreamer/elements/gsttensor_if.c:375]: thank you.

**Review by anyj0527 — APPROVED** [review]
LGTM 👍

**Review by songgot — APPROVED** [review]
LGTM

**Review by myungjoo — APPROVED** [review]

**Pull request merged**